pkg/api/vm/handler.go

```diff
@@ -717,6 +719,48 @@ func (h *vmActionHandler) createVMImages(templateVersion *harvesterv1.VirtualMac
 	return nil
 }
 
+// Since the LH volume creation and replica scheduling are asynchronous
+// Even the CreateVolume csi call success, the replica scheduling may still fail
+// However for the sc with VolumeBindingMode is Immediate
+// we could check annotation for corresponding error if it exists
+func (h *vmActionHandler) checkAttachable(pvc *corev1.PersistentVolumeClaim) error {
+	// Even Harvester specify default StorageClass,
+	// we check the sc existence for robustness
+	if pvc.Spec.StorageClassName == nil {
+		return fmt.Errorf("volme %v with empty StorageClass is not permitted", pvc.Name)
+	}
+
+	sc, err := h.storageClassCache.Get(*pvc.Spec.StorageClassName)
+	if err != nil {
+		return err
+	}
+
+	if sc.VolumeBindingMode == nil {
```

Comment:
as `sc.VolumeBindingMode` is not defined and controlled by this controller, would suggest to return nil rather than an error, log warning message instead

upstream may change, if we return error, it will block in certain cases

this feature is `try to alert user`, only return error when everything meets the condition

Reply:
Since during VolumeBindingMode is unset, "Immediate" mode is used by default, https://kubernetes.io/docs/concepts/storage/storage-classes/#volume-binding-mode. "VolumeBindingMode" never points to nil. How about keeping this checking for robustness as we discussed before? Thanks

Reply:
The source code reveals more. 

When a CSI driver sees `VolumeBindingMode` is nil in the SC, it uses `Immediate` as the value. But it does mean, the SC's `VolumeBindingMode` is never nil. The generated protocol buffer code, also checks !nil to do something.

vendor/k8s.io/api/storage/v1/types.go

```
type StorageClass struct {
...
	// VolumeBindingMode indicates how PersistentVolumeClaims should be
	// provisioned and bound.  When unset, VolumeBindingImmediate is used.
	// This field is only honored by servers that enable the VolumeScheduling feature.
	// +optional
	VolumeBindingMode *VolumeBindingMode `json:"volumeBindingMode,omitempty" protobuf:"bytes,7,opt,name=volumeBindingMode"`
...
```

```
vendor/k8s.io/api/storage/v1/generated.pb.go:	if m.VolumeBindingMode != nil {
..
vendor/k8s.io/api/storage/v1/generated.pb.go:	if m.VolumeBindingMode != nil {
..
```

Reply:
Update checking for StorageClass `VolumeBindingMode` based on suggestion. If this field point to nil, we just log warning message and return nil. The remaining handling works as fix for issue #1695